assets/js/state_template.js

```diff
@@ -0,0 +1,22 @@
+// Single state object
+var state = {
+    items: []
+};
+// State modification functions
+var addItem = function(state, item) {
+    state.items.push(item);
+};
+// Render functions
+var renderList = function(state, element) {
+    var itemsHTML = state.items.map(function(item) {
+        return '<li>' + item + '</li>';
+    });
+
+    element.html(itemsHTML);
+};
+// Event listeners
+$('.shopping-list-add').submit(function(event) {
+    event.preventDefault();
+    addItem(state, $('.shopping-list-add-input').val());
```

Comment:
You can alternatively, invoke (e).preventDefault(); -- less typing, less prone to typos (same with "function(event)"